src/SwiftPackage.ts

```diff
@@ -49,32 +49,45 @@ export interface Dependency {
 export class SwiftPackage implements PackageContents {
 	private constructor(
         readonly folder: vscode.WorkspaceFolder,
-        public contents?: PackageContents
+        private contents?: PackageContents|null
     ) {}
 
     public static async create(folder: vscode.WorkspaceFolder): Promise<SwiftPackage> {
+        let contents = await SwiftPackage.loadPackage(folder);
+        return new SwiftPackage(folder, contents);
+    }
+
+    public static async loadPackage(folder: vscode.WorkspaceFolder): Promise<PackageContents|null|undefined> {
         try {
-            let contents = await SwiftPackage.loadPackage(folder);
-            return new SwiftPackage(folder, contents);
+            const { stdout } = await exec('swift package describe --type json', { cwd: folder.uri.fsPath });
+            return JSON.parse(stdout);
         } catch(error) {
-            // TODO: output errors
-            return new SwiftPackage(folder, undefined);
+            const execError = error as {stderr: string};
+            // if caught error and it begins with "error: root manifest" then there is no Package.swift
+            if (execError.stderr.startsWith("error: root manifest")) {
+                return undefined;
+            } else {
+                // otherwise it is an error loading the Package.swift so return `null` indicating
+                // we have a package but we failed to load it
+                return null;
+            }
         }
     }
 
-    public static async loadPackage(folder: vscode.WorkspaceFolder): Promise<PackageContents> {
-        const { stdout } = await exec('swift package describe --type json', { cwd: folder.uri.fsPath });
-        return JSON.parse(stdout);
+    public async reload() {
+        this.contents = await SwiftPackage.loadPackage(this.folder);
     }
 
-    public async reload() {
-        try {
-            this.contents = await SwiftPackage.loadPackage(this.folder);
-        } catch {
-            this.contents = undefined;
-        }
+    // Return if Package.swift is valid
+    public get isValid(): boolean {
+        return this.contents !== null && this.contents !== undefined;
     }
 
+    // Did we find a Package.swift
+    public get foundPackage(): boolean {
+        return this.contents !== null;
```

Comment:
```suggestion
        return this.contents !== undefined;
```

as undefined = no package found

But do we even need that anywhere? Shouldn't all features that depend on packages require a valid package?